src/network/networkd-link.c

```diff
@@ -1039,6 +1054,24 @@ static int link_set_bridge_fdb(Link *const link) {
         return r;
 }
 
+static int link_set_proxy_arp(Link *const link) {
+        char buf[DECIMAL_STR_MAX(unsigned) + 1];
+        const char *p = NULL;
+        int r;
+
+        if (!link_proxy_arp_enabled(link))
+                return 0;
+
+        p = strjoina("/proc/sys/net/ipv4/conf/", link->ifname, "/proxy_arp");
+        xsprintf(buf, "%u", (unsigned) link->network->proxy_arp);
```

Comment:
this can only be 0 or 1, right?

If so, then maybe avoid using "buf" at all, and just use the existing `one_zero()` call instead?